Minio/Helper/RequestUtil.cs

```diff
@@ -0,0 +1,161 @@
+﻿
+/*
+ * Minio .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using Minio.Exceptions;
+using Minio.Helper;
+using System;
+using System.Text.RegularExpressions;
+
+namespace Minio
+{
+    public class RequestUtil
+    {
+        public static Uri getEndpointURL(string endPoint, bool secure)
+        {
+            if (endPoint.Contains(":"))
+            {
+                string[] parts = endPoint.Split(':');
+                string host = parts[0];
+                string port = (parts.Length > 1) ?  parts[1] : null; 
```

Comment:
`parts.Length > 1` will always be true as `endPoint` contains ':' here. This function can be refactored to perform the check about valid ip and valid host only once.